servant-server/test/Servant/ServerSpec.hs

```diff
@@ -505,7 +501,7 @@ responseHeadersSpec :: Spec
 responseHeadersSpec = describe "ResponseHeaders" $ do
   with (return $ serve (Proxy :: Proxy ResponseHeadersApi) responseHeadersServer) $ do
 
-    let methods = [(methodGet, 200), (methodPost, 201), (methodPut, 200), (methodPatch, 200)]
+    let methods = [(methodGet, 200), (methodPost, 200), (methodPut, 200), (methodPatch, 200)]
```

Comment:
This requires some refactoring of the test suite, which would be useful now in quite a few of the tests. I'm inclined to say that this should be part of the PR (we can considerably reduce the testing logic too now), but it'll take a while longer.